src/renderers/shared/fiber/ReactFiberScheduler.js

```diff
@@ -1387,12 +1392,17 @@ module.exports = function<T, P, I, TI, PI, C, CX, PL>(
   }
 
   function unbatchedUpdates<A>(fn: () => A): A {
+    const previousIsUnbatchingUpdates = isUnbatchingUpdates;
     const previousIsBatchingUpdates = isBatchingUpdates;
+    // This is only true if we're nested inside either batchedUpdates or a
+    // lifecycle method.
+    isUnbatchingUpdates = isBatchingUpdates || isPerformingWork;
```

Comment:
You're right, this check is redundant. I added this check first, then realized it made more sense to check `isPerformingWork` in `scheduleUpdate`, but forgot to remove the check here. I'll remove it.